### PR TITLE
feat: enrich webhook scanner panic unexpected errors

### PR DESCRIPTION
**Pull request by shalev007**
- feat: add state class
- fix: add state intialization
- feat: add cluster name to webhook
- feat: use custom cli client
- fix: test use custom cli client
- feat: go mod tidy
- feat: add state as a dependency injection
- feat: remove unnecessary panic
- feat: reuse state variables
- test: fix test dependencies
- fix: wrong client issue
- feat: remove duplicated cli-service-client
- feat: remove duplicated import
- feat: make validation service dependency public
- feat: remove duplicated cliClient
- feat: remove unnecessary dependency
- feat: delete unnecessary new function

### CODE REVIEW

**Review by royhadad — COMMENTED** [review]
did you add the webhook version? it's the most important thing

File: pkg/serviceState/serviceState.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+func New() *ServiceState {
+
+	return &ServiceState{
+		clientId:       shortuuid.New(),
+		serviceType:    WEBHOOK,
+		token:          os.Getenv(enums.Token),
+		clusterName:    os.Getenv(enums.ClusterName),
+		policyName:     os.Getenv(enums.Policy),
+		isEnforceMode:  os.Getenv(enums.Enforce) == "true",
+		serviceVersion: config.WebhookVersion,
```
Comment (shalev007) [inline pkg/serviceState/serviceState.go:37]: @royhadad

Comment (royhadad) [inline pkg/serviceState/serviceState.go:37]: no need for this to be generic, being specific makes it more readable
```suggestion
		webhookVersion: config.WebhookVersion,
```

File: main.go [hunk 2]
```diff
@@ -61,20 +64,39 @@ func start(port string) {
...
 
-	validationController := controllers.NewValidationController(errorReporter, k8sMetadataUtilInstance)
+	validationController := controllers.NewValidationController(basicCliClient, state, errorReporter, k8sMetadataUtilInstance)
 	healthController := controllers.NewHealthController()
 	// set routes
 	http.HandleFunc("/validate", validationController.Validate)
 	http.HandleFunc("/health", healthController.Health)
 	http.HandleFunc("/ready", healthController.Ready)
 
+	initMetadataLogsCronjob(validationController.ValidationService)
```
Comment (TzlilSwimmer123) [inline main.go:98]: Maybe a comment will be useful for other developers?

File: pkg/clients/cliServiceClient.go [hunk 3]
```diff
@@ -225,3 +226,27 @@ type ReportK8sMetadataRequest struct {
...
+	ClusterName    string       `json:"clusterName"`
+	K8sVersion     string       `json:"k8sVersion"`
+	PolicyName     string       `json:"policyName"`
+	IsEnforceMode  bool         `json:"isEnforceMode"`
+	ServiceVersion string       `json:"serviceVersion"`
+	ErrorMessage   string       `json:"errorMessage"`
+	StackTrace     string       `json:"stackTrace"`
+}
+
+func (c *CliClient) ReportError(reportCliErrorRequest ReportErrorRequest, uri string) (StatusCode int, Error error) {
```
Comment (TzlilSwimmer123) [inline pkg/clients/cliServiceClient.go:243]: rename `reportCliErrorRequest `

File: pkg/errorReporter/errorReporter.go [hunk 4]
```diff
@@ -37,16 +36,21 @@ func (reporter *ErrorReporter) ReportUnexpectedError(unexpectedError error) {
...
+		IsEnforceMode:  reporter.state.GetIsEnforceMode(),
+		ServiceVersion: reporter.state.GetServiceVersion(),
+		ErrorMessage:   errorMessage,
+		StackTrace:     string(debug.Stack()),
 	}, uri)
 
 	if err != nil {
 		// using fmt.Println instead of logger to avoid circular dependency
-		fmt.Println(fmt.Sprintf("ReportError status code: %d, err: %s", statusCode, err.Error()))
+		fmt.Printf("ReportError status code: %d, err: %s", statusCode, err.Error())
```
Comment (TzlilSwimmer123) [inline pkg/errorReporter/errorReporter.go:54]: 💯

File: pkg/services/validationService.go [hunk 5]
```diff
@@ -56,29 +51,8 @@ type Metadata struct {
...
-	}
-}
-
-func NewValidationServiceWithCustomCliServiceClient(cliServiceClient *cliClient.CliClient, k8sMetadataUtilInstance *k8sMetadataUtil.K8sMetadataUtil, errorReporter *errorReporter.ErrorReporter) *ValidationService {
-	return &ValidationService{
-		CliServiceClient: cliServiceClient,
-		K8sMetadataUtil:  k8sMetadataUtilInstance,
-		errorReporter:    errorReporter,
-	}
+	ErrorReporter    *errorReporter.ErrorReporter
```
Comment (TzlilSwimmer123) [inline pkg/services/validationService.go:54]: 💯

**Review by TzlilSwimmer123 — APPROVED** [review]
Left couple of comments 💪

**Pull request merged**